packages/jest-runner/src/runTest.ts

```diff
@@ -104,9 +105,35 @@ async function runTestInternal(
   }
 
   const transformer = new ScriptTransformer(config);
-  const TestEnvironment: typeof JestEnvironment = interopRequireDefault(
-    transformer.requireAndTranspileModule(testEnvironment),
-  ).default;
+  let TestEnvironment: typeof JestEnvironment;
+  try {
+    TestEnvironment = interopRequireDefault(
+      transformer.requireAndTranspileModule(testEnvironment),
+    ).default;
+  } catch (err) {
+    try {
```

Comment:
you should check if the error is `if (error.code === 'ERR_REQUIRE_ESM') {` before attempting `import`

Reply:
@SimenB sorry about the slow response as well. Let me know if my change works. I just put a simple `if`, otherwise throw the error